packages/core/src/sync/sync-api.ts

```diff
@@ -265,6 +265,10 @@ export class SyncApi implements ISyncApi {
     })
   }
 
+  async syncComplete(model: string): Promise<boolean> {
+    return !this.modelsToSync.has(model)
```

Comment:
Unfortunately figuring out which models are currently syncing is a bit more complicated. There are two types of syncs: 1. continuous sync, 2. historic sync. 

We want to prevent queries if there are historic syncs going on for that model. The continuous sync is always running for all indexed models

Reply:
Any ideas on how we could test this change effectively so a nuance like this would be caught?

Reply:
We can check the active/pending jobs and see if the models is included, we can keep a running list of models that having pending history syncs and update when syncs are done. We can have a table with previous sync information for each model.

Reply:
Easiest thing to do is to not allow ANY queries if history syncs are running. So basically if there are any active or pending history sync jobs

Reply:
I think disallowing all queries could cause issues.  Imagine an app has been running a while with some set of indexed models.  Now it wants to start using a new model, so it adds that model and starts the historical sync, but doesn't update the app logic to use the new model until the sync is complete.  But now the whole app will fail for the queries on all the existing models that were already working.